File: spring-java-starter/spring-java-mysql/starter/src/main/java/io/github/mrgsrylm/entity/UserEntity.java
package io.github.mrgsrylm.entity;

import io.github.mrgsrylm.enums.TokenClaims;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import io.github.mrgsrylm.enums.Role;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "USERS")
public class UserEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String fullName;

    @Column(unique = true)
    private String username;

    @Column(unique = true)
    private String email;

    private String password;

    @Enumerated(EnumType.STRING)
    @Column(name = "ROLE")
    private Role role;

    public Map<String, Object> getClaims() {
        final Map<String, Object> claims = new HashMap<>();
        claims.put(TokenClaims.ID.getValue(), this.id);
        claims.put(TokenClaims.USERNAME.getValue(), this.username);
        claims.put(TokenClaims.ROLES.getValue(), List.of(this.role));
        claims.put(TokenClaims.USER_FULL_NAME.getValue(), this.fullName);
        claims.put(TokenClaims.EMAIL.getValue(), this.email);
        return claims;
    }
}

File: spring-kotlin-starter/spring-kotlin-mysql/starter/src/main/kotlin/io/github/mrgsrylm/entity/UserEntity.kt
package io.github.mrgsrylm.entity

import io.github.mrgsrylm.enums.Role
import io.github.mrgsrylm.enums.TokenClaims
import jakarta.persistence.*
import kotlin.reflect.jvm.internal.impl.load.kotlin.JvmType

@Entity
@Table(name = "USERS")
class UserEntity(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long,

        var fullName: String,

        @Column(unique = true)
        var username: String,

        @Column(unique = true)
        var email: String,

        var password: String,

        @Enumerated(EnumType.STRING)
        @Column(name = "ROLE")
        var role: Role
) : BaseEntity(){

    fun getClaims(): Map<String, Any> {
        val claims = HashMap<String, Any>();
        claims[TokenClaims.ID.value] = this.id
        claims[TokenClaims.USER_FULL_NAME.value] = this.fullName
        claims[TokenClaims.USERNAME.value] = this.username
        claims[TokenClaims.EMAIL.value] = this.email
        claims[TokenClaims.ROLES.value] = this.role

        return claims;
    }

}